tiered-storage/jcloud/src/main/java/org/apache/bookkeeper/mledger/offload/jcloud/impl/BlobStoreBackedReadHandleImpl.java

```diff
@@ -55,6 +57,8 @@ public class BlobStoreBackedReadHandleImpl implements ReadHandle {
     private final BackedInputStream inputStream;
     private final DataInputStream dataStream;
     private final ExecutorService executor;
+    // this Map is accessed only by one thread
+    private final Map<Long, Long> entryOffsets = new TreeMap<>();
```

Comment:
The map won't expire any items until the `ReadHandle` is closed. It may cost a lot of memory. Could we introduce an expiration policy?

Reply:
yes, I was thinking about that as well.
I will sure the usual Guava cache.
good point !

Reply:
I am not using a cache with a 10 minutes TTL.
it should be enough to cover the cases in which the Subscription reads old entries